File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/services/parameters/ServicesListParameters.java
package co.uk.swarmbit.docker.api.services.parameters;

import co.uk.swarmbit.util.EncoderDecoder;
import co.uk.swarmbit.docker.api.common.parameters.FiltersParameters;
import co.uk.swarmbit.rest.QueryParam;

import java.util.Optional;

public class ServicesListParameters implements FiltersParameters {

    private static final String FILTERS_NAME = "filters";

    private Optional<QueryParam> filters;

    public ServicesListParameters() {
        filters = Optional.empty();
    }

    public Optional<QueryParam> getFilters() {
        return filters;
    }

    public ServicesListParameters setFilters(ServicesFilters filters) {
        if (filters != null) {
            this.filters = Optional.of(new QueryParam(FILTERS_NAME, EncoderDecoder.jsonUrlEncode(filters.getJson())));
        }
        return this;
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/services/parameters/ServicesListParameters.kt
package com.swarmbit.docker.api.services.parameters

import com.swarmbit.docker.api.common.annotation.DockerRemoteApiMinVersion
import com.swarmbit.docker.api.common.parameters.FiltersParameters
import com.swarmbit.docker.api.common.rest.model.QueryParam

class ServicesListParameters : FiltersParameters {

    companion object {
        private const val FILTERS_NAME = "filters"
        private const val STATUS_NAME = "status"
    }

    private var innerFilters: QueryParam? = null

    fun setFilters(filters: ServicesFilters): ServicesListParameters {
        this.innerFilters = QueryParam(FILTERS_NAME, filters.asUrlJson())
        return this
    }

    @DockerRemoteApiMinVersion("v1.41")
    fun setStatus(status: Boolean): ServicesListParameters {
        this.innerFilters = QueryParam(STATUS_NAME, status.toString())
        return this
    }

    override val filters: QueryParam?
        get() = innerFilters

}